### ISSUE TITLE
what is the use of "len_tgt_list" ,cond_span_l_loss,cond_span_r_loss?

### ISSUE THREAD

**Issue by Sandy26** [donglixp/coarse2fine#8]
Hi Li,
I understand the use of cond_span_l and con_span_r but doesn't the loss get calculated for these variables during training? Then why do we need cond_span_l_loss and cond_span_r_loss fields?

Also, is the variable len_tgt_list (declared in wikisql/table/IO.py line 181) useful? I see it declared but I cannot see it used anywhere.
Thank you,
Shruti

**Comment by KeSyren** [comment]
Hi 
Could anyone tell me the PyTorch version of this project? I've got the error when running the code. 

![image](https://user-images.githubusercontent.com/6985499/43049080-d713830a-8dbf-11e8-9e07-4b6783b562fd.png)

**Comment by donglixp** [comment]
@KeSyren Please find the PyTorch version in https://github.com/donglixp/coarse2fine/blob/master/README.md .

**Comment by donglixp** [comment]
@Sandy26 

cond_span_l_loss and cond_span_r_loss are gold values that are used to compute loss during training.

https://github.com/donglixp/coarse2fine/blob/8d135d04ce16107e913028212c4df615a3940d7f/wikisql/table/Trainer.py#L112

**Comment by donglixp** [comment]
@Sandy26 

len_tgt_list is defined to get statistics about examples. It's not used during training and inference.
